x-pack/legacy/plugins/ml/public/application/jobs/jobs_list/components/utils.js

```diff
@@ -147,49 +148,53 @@ function showResults(resp, action) {
   }
 }
 
-export function cloneJob(jobId) {
-  loadFullJob(jobId)
-    .then((job) => {
-      if(job.custom_settings && job.custom_settings.created_by) {
-        // if the job is from a wizards, i.e. contains a created_by property
-        // use tempJobCloningObjects to temporarily store the job
-        mlJobService.tempJobCloningObjects.job = job;
-
-        if (
-          job.data_counts.earliest_record_timestamp !== undefined &&
-          job.data_counts.latest_record_timestamp !== undefined &&
-          job.data_counts.latest_bucket_timestamp !== undefined) {
-          // if the job has run before, use the earliest and latest record timestamp
-          // as the cloned job's time range
-          let start = job.data_counts.earliest_record_timestamp;
-          let end = job.data_counts.latest_record_timestamp;
-
-          if (job.datafeed_config.aggregations !== undefined) {
-            // if the datafeed uses aggregations the earliest and latest record timestamps may not be the same
-            // as the start and end of the data in the index.
-            const bucketSpanMs = parseInterval(job.analysis_config.bucket_span).asMilliseconds();
-            // round down to the start of the nearest bucket
-            start = Math.floor(job.data_counts.earliest_record_timestamp / bucketSpanMs) * bucketSpanMs;
-            // use latest_bucket_timestamp and add two bucket spans minus one ms
-            end = job.data_counts.latest_bucket_timestamp + (bucketSpanMs * 2) - 1;
-          }
-
-          mlJobService.tempJobCloningObjects.start = start;
-          mlJobService.tempJobCloningObjects.end = end;
+export async function cloneJob(jobId) {
+  try {
+    const job = await loadFullJob(jobId);
+    if (job.custom_settings && job.custom_settings.created_by) {
```

Comment:
If you add custom URLs using the Edit job functionality in the Jobs list, the `created_by` setting for the wizard is removed (as you used not to be able to edit custom URLs in the wizards). We should now retain the `created_by` setting when editing an existing job.

Reply:
https://github.com/elastic/kibana/pull/51281/commits/8e122c8ad5c9f320fa4c05db3be7f72e3ee61002